th-data-split.html

```diff
@@ -21,65 +23,118 @@
-            </template>
-
+    <!--core-collapse shows table settings. Toggled off by default-->
+    <core-collapse id="ctrl_collapse" opened="false">
+      <p class="info">Apply a filter on a column to narrow down the results.</p>
+      <label for="func_selector">Filter on column:</label>
+      <core-selector id="value_selector" selected="{{splitCol}}" valueattr="label">
+        <template repeat="{{attr in _dataAttributes}}">
+          <div class="col" label="{{attr}}">{{attr}}</div>
+        </template>
+      </core-selector>
+      <label>split by:</label>
+      <input is="core-input" value="{{splitChar}}">
+      <label>New column labels:</label>
+      <template repeat="{{name, index in splitColNames}}">
+        <input is="core-input" value="{{name}}" _index="{{index}}" class="col-name" on-change="{{_splitNameChanged}}" >
+      </template>
+    </core-collapse>
 
   </template>
   <script>
     Polymer({
+
+      /**
+       * Input data
+       *
+       * @property input
+       * @type object
+       * @default undefined
+       */
+
+      /**
+       * Output data
+       *
+       * @property output
+       * @type object
+       * @default undefined
+       */
+
+      /**
+       * Split character. Defaults to ',''
+       *
+       * @property splitChar
+       * @type string
+       * @default ,
+       */
       splitChar: ',',
-      splitColumnNames: [],
+
+      /**
+       * Name of the original column that will be split.
+       *
+       * @property splitCol
+       * @type array
+       * @default undefined
+       */
+      splitCol: [],
+
+      /**
+       * Names of the new columns generated after the split.
+       *
+       * @property splitColNames
+       * @type array
+       * @default []
+       */
+      splitColNames: [],
+
+
+      /**
+       *  Fired when this polymer element is ready
+       *  
+       *  @event ready
+       */
       ready: function () {
-        //open the collapse if it the component is included
+        //open the collapse if the component is included
         if(this.included && !this.$.ctrl_collapse.opened) {
           this.$.ctrl_collapse.toggle();
         }
       },
-      inputChanged: function() {
-
-          var self = this;
-          var firstItem = self.input[0];
-          if(!firstItem) {
-            return;
-          }
-          var attributes = Object.keys(firstItem);
-          
-          if(JSON.stringify(self._dataAttributes) === JSON.stringify(attributes) ) {
-            //self._mapChartData();
-          }
-          else {
-            self._dataAttributes = attributes;
-          }
 
-          self._calculateOutput();          
+      /**
+       *  Fired automatically when the input is changed.
+       *  This event parses input data, then
+       *  calls a private event to recalculate output.
+       *  
+       *  @event inputChanged
```

Comment:
I was browsing Polymer component documentations and I noticed that they don't document methods unless the method is supposed to be called from outside (example [here](https://github.com/Polymer/core-selector/blob/master/core-selector.html))
And in case they want some sort of documentation the use // comments so no JSDoc is generated for those methods. So we can follow the same pattern and remove the function docs. Sorry for not researching it earlier.